### ISSUE TITLE
No member named 'getTypeAsWritten' in 'clang::ClassTemplateSpecializationDecl' in upstream latest clang

### ISSUE THREAD

**Issue by gamesh411** [include-what-you-use/include-what-you-use#1538]
When compiling with the latest llvm-main/main, the following compilation error occurs:
```
.../include-what-you-use/iwyu.cc:4080:38: error: no member named 'getTypeAsWritten' in 'clang::ClassTemplateSpecializationDecl'
4080 |       const TypeLoc type_loc = decl->getTypeAsWritten()->getTypeLoc();
```
I suspect this is caused by a recently landed PR: https://github.com/llvm/llvm-project/pull/91393
This is a re-application of this change: https://github.com/llvm/llvm-project/pull/81642
The interface of template specializations is changed.

**Comment by kimgr** [comment]
Thanks, I'm trying to wrinkle my head around this.

**Comment by bolshakov-a** [comment]
It really appears to be slightly tricky and requires some refactoring. I'm working on this too.

**Comment by kimgr** [comment]
@bolshakov-a Glad to hear it, I had just about given up 🥲.

**Comment by bolshakov-a** [comment]
Hey @kimgr, this can be closed now.

**Comment by kimgr** [comment]
Fixed on master as of dc9fd2c258a9f29881d73330c9b6fdb13b41442c.
